### ISSUE TITLE
Use % and REM instead of PX to set font sizes

### ISSUE THREAD

**Issue by fduraibi** [RocketChat/Rocket.Chat#1607]
Add an option in the settings to allow the user to increase the font size. Different screen resolutions and different fonts/languages can make it hard to read small fonts.
##

**Comment by engelgabriel** [comment]
+1

**Comment by internalfx** [comment]
+1

**Comment by fduraibi** [comment]
Also allowing the admin to change the font itself would be great.

**Comment by fduraibi** [comment]
I have been working on this and I added font selector in the  settings page and can store the selected font into the user's profile. However, I have few problems with the most important part which is changing the actual font size.

- [ ] Which text to adjust the size of and which to keep as is (messages, settings, side menu ...).
- [ ] Make sure icons that are from fonts do not get resized.
- [ ] Should we save the font setting in the server so it will apply to all user's sessions, or keep it local so the user can adjust the setting based on the device specs (e.g. on phone vs desktop vs tablet and different screen resolutions)
- [ ] Based on my searches many developers recommend using percentage and `em` instead of `px` to set font sizes in style sheets, that makes the dynamic re-sizing much easier with consistent results. So should we redo the font sizing in the current stylesheets?
- [ ] Which way to use to access the html elements? 
    - Using element ID e.g. ("#messagetext").css("font-size", newsize)
    - or Adding css class to element ("#messagetext").addClass(newsizeClass)
    - Currently most text elements do not have IDs, so we can use element tags and existing classes but I am not sure if it is reliable, or we add IDs. Adding IDs will also make it easy to control the sizing of different elements e.g. User's name, messages, system notifications, side-panel ...etc

I need your comments and suggestions

**Comment by internalfx** [comment]
@fduraibi I have typically handled font sizes like so....

```css
html {
  font-size: 62.5%; /* After this you can now use rem (relative em). */
}

// example
h1 {
  font-size: 2.4rem; /* 24px font */
}
```

`1rem` will be a 10px font. `1.6rem` will be a 16px font. This makes it easy to get exact results.

I hope that is helpful.

**Comment by fduraibi** [comment]
@internalfx yes, this is one of the most popular solutions for dynamic font sizing, i will give it a try and see the result

**Comment by fduraibi** [comment]
For saving the settings I think it should be local only. Since when I tested it on my phone the normal font size and even small size looked really good and readable, while on the desktop screen I needed a larger font size.

**Comment by fduraibi** [comment]
Here is the current progress: https://github.com/fduraibi/Rocket.Chat/tree/resizable-font

It kinda works fine except in some areas specially with firefox.

For example:
![rocket_font](https://cloud.githubusercontent.com/assets/432460/11945645/287c955c-a820-11e5-97f3-bd3327de2fae.png)
The `textarea` height is calculated i think by `packages/rocketchat-ui/lib/textarea-autogrow.js` and I couldn't figure out how to get it to do the calculation based on the new change.

Some stuff require adjusting the paddings/margins so it would fit the bigger font. It just needs more work on the `base.css`.

**Comment by fduraibi** [comment]
Screenshots of the current progress in different font sizes: (On Firefox under Linux)

### Small
![rocket_font-small](https://cloud.githubusercontent.com/assets/432460/11946013/0d98bb68-a824-11e5-9f78-ad07bbceac7d.png)
----

### Medium (the default and just like the original)
![rocket_font-medium](https://cloud.githubusercontent.com/assets/432460/11946014/123f48bc-a824-11e5-8a76-81c92fa455b0.png)
----
### Large
![rocket_font-large](https://cloud.githubusercontent.com/assets/432460/11946018/1a048db4-a824-11e5-8f3e-13b756ddbd2f.png)
----
### Very Large
![rocket_font-vvlarge](https://cloud.githubusercontent.com/assets/432460/11946020/1e1497c8-a824-11e5-89f9-defb7d5fdca0.png)

**Comment by engelgabriel** [comment]
Looks good! 

PS. If this is going to be a local setting, what would be the difference than setting up the "zoom" on my browser?

**Comment by fduraibi** [comment]
I was trying to avoid enlarging everything. In hope to keep as much of space for the messages themselves and not be consumed by the side menu. So at first I wanted to only enlarge the message box only since everything else (like room names, menus ...etc) are mostly static and you don't need to re-read them, however, resizing just the messages means I also need to resize the textarea and create separate styles that won't affect other elements and that needed a lot of changes.

But I now feel it is just easier to use the browsers zoom

**Comment by engelgabriel** [comment]
I understand, but seem like a lot of work for not much return :)

We should always make sure that the browser's zoom works well. You are right about using percentage and em instead of px to set font sizes in style sheets, that makes the dynamic resizing much easier with consistent results. So should we redo the font sizing in the current stylesheets indeed.

**Comment by fduraibi** [comment]
Yeah, I totally agree with you. And the browsers zoom works well. So I will close this for now, and we can open it if someone thinks otherwise.

**Comment by engelgabriel** [comment]
Lets re-purpose the issue.

**Comment by rogerpfaff** [comment]
To repurpose the issue I would like to mention all those people lacking in visual ability by age alone.

The visibiliy on a smartphone below 5 inch is really poor in standard settings and to convince my coworkers with glasses in different forms and shapes it would be a selling point to direct them to the settings page for better readability.

I'm sure any work done into this issue will be worth a wider range of users.

**Comment by engelgabriel** [comment]
@ggazzo what is the progress on this? Have we managed to remove all the PX to set font sizes?

**Comment by dougfabris** [comment]
Hi there, thanks for the contribution! 🚀 💯 

We've been working a lot to improve our UI and make it accessible for everyone. This is an old issue and we now have a fully new experience, our sizes are now relatives so it's way easier to increase or decrease font sizes. Also, there is a dedicated page under the user account to select some accessibility configurations such as font size. 

---

**Questions? Help needed? Feature Requests?** 

- Join our [Open Server](https://open.rocket.chat/) in the #support channel and feel free to raise a question
- Join our [Community Forum](https://forums.rocket.chat/) and search/create a post there
